viewer/src/app/feature-service.service.ts

```diff
@@ -96,9 +98,42 @@ export type featureCollectionGeoJSON = {
   numberReturned?: number
 }
 
+export type DataProjectionMapping = {
+  dataProjection: ProjectionLike //Projection of the data we are reading
+  visualProjection: ProjectionLike //Projection of the feature geometries created by this function
+}
+
 export type DataUrl = {
   url: string
-  projection: ProjectionLike
+  dataMapping: DataProjectionMapping
+}
+export const defaultMapping: DataProjectionMapping = { dataProjection: 'EPSG:4326', visualProjection: 'EPSG:3857' }
+
+export function projectionAttribute(value: ProjectionLike = 'HTTP://WWW.OPENGIS.NET/DEF/CRS/OGC/1.3/CRS84'): DataProjectionMapping {
+  //console.log(typeof value)
+
+  initProj4()
+
+  if (value) {
+    if (typeof value === 'string') {
+      //console.log(value.substring(value.lastIndexOf('/') + 1).toLowerCase())
+      if (value.substring(value.lastIndexOf('/') + 1).toLocaleUpperCase() === 'CRS84') {
+        //'EPSG:3857' //Default the map is in Web Mercator(EPSG: 3857), the actual coordinates used are in lat-long (EPSG: 4326)
+        //console.log('map')
+        return defaultMapping
+      }
+      if (value.toUpperCase().startsWith('HTTP://WWW.OPENGIS.NET/DEF/CRS/EPSG/')) {
+        const projection = 'EPSG:' + value.substring(value.lastIndexOf('/') + 1)
+        return { dataProjection: projection, visualProjection: 'EPSG:3857' }
+      }
+      return { dataProjection: value, visualProjection: value }
+    } else {
+      console.log('wrong value ')
```

Comment:
kan in 1 log statement. Zou tevens console.error gebruiken ipv console.log

Reply:
volgens mij niet als je het als apart argument meegeeft, bijv:

```
const obj = {};
obj.prop = 123;
console.log('hi', obj);
```

dan krijg je obj netjes als object in de console niet als [object] string.